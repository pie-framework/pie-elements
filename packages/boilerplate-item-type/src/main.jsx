import React from 'react';
import PropTypes from 'prop-types';
import { PreviewPrompt } from '@pie-element/pie-utils/render-ui';
import UTILS from '@pie-element/pie-utils';

const {test, callTest} = UTILS;

callTest('player');
export default class Main extends React.Component {
  static propTypes = {
    model: PropTypes.object,
    session: PropTypes.object,
    onSessionChange: PropTypes.func.isRequired,
  };

  static defaultProps = {};

  render() {
    const { model } = this.props;

    return (
      <div>
        {model.prompt && (
          <div>
            <p>{test}</p>
            <p>Boilerplate Item</p>
            <PreviewPrompt prompt={model.prompt}/>
          </div>
        )}
      </div>
    );
  }
}
