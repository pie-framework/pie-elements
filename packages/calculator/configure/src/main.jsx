import React from 'react';
import PropTypes from 'prop-types';
import { layout, TwoChoice } from '@pie-lib/config-ui';
import { CalculatorLayout } from '@pie-element/calculator';
import Typography from '@material-ui/core/Typography';

const MainTypes = {
  model: PropTypes.object.isRequired,
  onChange: PropTypes.func.isRequired,
};

export class Main extends React.Component {
  static propTypes = MainTypes;

  onModeChange = (mode) => {
    const { model, onChange } = this.props;
    const update = { ...model, mode };

    onChange(update);
  };

  render() {
    const { model } = this.props;

    return (
      <layout.ConfigLayout hideSettings={true} settings={null}>
        <TwoChoice
          header="Choose a Calculator"
          value={model.mode}
          onChange={this.onModeChange}
          one={{ label: 'Basic', value: 'basic' }}
          two={{ label: 'Scientific', value: 'scientific' }}
        />
        <br />

        <CalculatorLayout mode={model.mode} onClose={() => ({})} />
        <br />

        <Typography style={{ paddingTop: '4px' }}>
          Please note that the calculators are tools for students and do not record answers.
        </Typography>
      </layout.ConfigLayout>
    );
  }
}

export default class Stateful extends React.Component {
  static propTypes = MainTypes;

  constructor(props) {
    super(props);
    this.state = { model: props.model };
  }

  UNSAFE_componentWillReceiveProps(props) {
    this.setState({ model: props.model });
  }

  onChange = (model) => {
    this.setState({ model }, () => {
      this.props.onChange(this.state.model);
    });
  };

  render() {
    return <Main model={this.state.model} onChange={this.onChange} />;
  }
}
