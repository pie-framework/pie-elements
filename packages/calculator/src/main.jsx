import React from 'react';
import PropTypes from 'prop-types';
import DraggableCalculator from './draggable-calculator';
import CalculatorIcon from './calculator-icon';
import { withStyles } from '@material-ui/core/styles';
import classNames from 'classnames';

const styles = {
  icon: {
    transition: 'fill 200ms',
    '&:hover': {
      fill: 'black',
    },
    cursor: 'pointer',
    verticalAlign: 'middle',
    fill: 'grey',
  },
  active: {
    fill: 'black',
  },
};

class Main extends React.Component {
  static propTypes = {
    model: PropTypes.object.isRequired,
    classes: PropTypes.object.isRequired,
  };
  constructor(props) {
    super(props);
    this.state = { show: false };
  }

  onToggleShow = () => this.setState({ show: !this.state.show });
  onClickClose = () => this.setState({ show: false });

  render() {
    const { show } = this.state;
    const { mode } = this.props.model;
    const { classes } = this.props;
    return (
      <div>
        <CalculatorIcon
          className={show ? classNames(classes.icon, classes.active) : classes.icon}
          onClick={() => this.onToggleShow()}
        />
        <DraggableCalculator mode={mode} show={show} onClose={this.onClickClose} />
      </div>
    );
  }
}

export default withStyles(styles)(Main);
