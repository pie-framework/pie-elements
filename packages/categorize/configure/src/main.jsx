import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import classNames from 'classnames';
import Design from './design';

export class Main extends React.Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    className: PropTypes.string,
    model: PropTypes.object.isRequired,
    onModelChanged: PropTypes.func.isRequired,
    imageSupport: PropTypes.object
  };

  static defaultProps = {};

  render() {
    const {
      classes,
      className,
      model,
      onModelChanged,
      configuration,
      onConfigurationChanged
    } = this.props;

    return (
      <div className={classNames(classes.main, className)}>
        <Design
          imageSupport={this.props.imageSupport}
          title="Design"
          model={model}
          configuration={configuration}
          onChange={onModelChanged}
          onConfigurationChanged={onConfigurationChanged}
        />
      </div>
    );
  }
}
const styles = theme => ({
  main: {}
});

export default withStyles(styles)(Main);
