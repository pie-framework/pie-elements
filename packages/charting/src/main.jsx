import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import { color, Collapsible, hasText } from '@pie-lib/render-ui';
import { Chart, chartTypes } from '@pie-lib/charting';
import isEqual from 'lodash/isEqual';
import CorrectAnswerToggle from '@pie-lib/correct-answer-toggle';
import {GraphContainer} from '@pie-lib/graphing';

export class Main extends React.Component {
  static propTypes = {
    classes: PropTypes.object,
    model: PropTypes.object.isRequired,
    onAnswersChange: PropTypes.func,
    categories: PropTypes.array,
  };

  static defaultProps = { classes: {} };

  constructor(props) {
    super(props);

    this.state = {
      categories: props.categories || props.model.data,
      showingCorrect: false
    };
  }

  UNSAFE_componentWillReceiveProps(nextProps) {
    const { model: { data: nextData = [] } = {} } = nextProps;
    const { model: { data = [] } = {} } = this.props;

    if (!isEqual(nextData, data)) {
      this.setState({ categories: nextData });
    }
  }

  changeData = (data) =>
    this.setState(
      {
        categories: data,
      },
      () => this.props.onAnswersChange(data)
    );

  toggleCorrect = showingCorrect => this.setState({ showingCorrect });

  render() {
    const { categories, showingCorrect } = this.state;
    const { model, classes } = this.props;
    const {
      teacherInstructions,
      prompt,
      chartType,
      size,
      domain,
      range,
      title,
      editCategoryEnabled,
      addCategoryEnabled,
      categoryDefaultLabel,
      rationale,
      correctedAnswer,
      correctAnswer
    } = model;

    const correctData = correctAnswer && correctAnswer.data ? correctAnswer.data.map(data => {
      return {
        ...data,
        interactive: false,
        editable: false
      };
    }) : [];

    const showToggle = correctData && correctData.length > 0;

    return (
      <div className={classes.mainContainer}>
        <CorrectAnswerToggle
          show={showToggle}
          toggled={showingCorrect}
          onToggle={this.toggleCorrect}
        />

        {teacherInstructions && hasText(teacherInstructions) && (
          <React.Fragment>
            <Collapsible
              labels={{
                hidden: 'Show Teacher Instructions',
                visible: 'Hide Teacher Instructions',
              }}
            >
              <div dangerouslySetInnerHTML={{ __html: teacherInstructions }} />
            </Collapsible>
            <br />
          </React.Fragment>
        )}

        {prompt && (
          <React.Fragment>
            <div
              className={classes.prompt}
              dangerouslySetInnerHTML={{ __html: prompt }}
            />
            <br />
          </React.Fragment>
        )}

        {(showingCorrect && showToggle) ? (
          <Chart
            chartType={chartType}
            size={size}
            domain={domain}
            range={range}
            charts={[
              chartTypes.Bar(),
              chartTypes.Histogram(),
              chartTypes.LineDot(),
              chartTypes.LineCross(),
              chartTypes.DotPlot(),
              chartTypes.LinePlot(),
            ]}
            data={correctData || categories}
            title={title}
            onDataChange={this.changeData}
            editCategoryEnabled={false}
            addCategoryEnabled={false}
            categoryDefaultLabel={categoryDefaultLabel}
          />
        ) : (
          <Chart
            chartType={chartType}
            size={size}
            domain={domain}
            range={range}
            charts={[
              chartTypes.Bar(),
              chartTypes.Histogram(),
              chartTypes.LineDot(),
              chartTypes.LineCross(),
              chartTypes.DotPlot(),
              chartTypes.LinePlot(),
            ]}
            data={correctedAnswer || categories}
            title={title}
            onDataChange={this.changeData}
            editCategoryEnabled={editCategoryEnabled}
            addCategoryEnabled={addCategoryEnabled}
            categoryDefaultLabel={categoryDefaultLabel}
          />)}

        <br />
        {rationale && hasText(rationale) && (
          <Collapsible
            labels={{ hidden: 'Show Rationale', visible: 'Hide Rationale' }}
          >
            <div dangerouslySetInnerHTML={{ __html: rationale }} />
          </Collapsible>
        )}
      </div>
    );
  }
}

const styles = theme => ({
  mainContainer: {
    padding: theme.spacing.unit,
    color: color.text(),
    backgroundColor: color.background(),
  },
  prompt: {
    verticalAlign: 'middle',
  },
});

export default withStyles(styles)(Main);
