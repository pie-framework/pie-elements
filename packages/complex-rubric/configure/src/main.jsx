import React from 'react';
import PropTypes from 'prop-types';
import { RUBRIC_TYPES } from '@pie-lib/pie-toolbox/rubric';
import { layout } from '@pie-lib/pie-toolbox/config-ui';
import Radio from '@material-ui/core/Radio';
import RadioGroup from '@material-ui/core/RadioGroup';
import { withStyles } from '@material-ui/core/styles';
import { FormControlLabel } from '@material-ui/core';

const styles = {};
const rubricLabels = {
  [RUBRIC_TYPES.MULTI_TRAIT_RUBRIC]: 'Multi Trait Rubric',
  [RUBRIC_TYPES.SIMPLE_RUBRIC]: 'Simple Rubric',
  [RUBRIC_TYPES.RUBRICLESS]: 'Rubricless',
};

export class Main extends React.Component {
  static propTypes = {
    classes: PropTypes.object,
    canUpdateModel: PropTypes.bool,
    configuration: PropTypes.object,
    model: PropTypes.object,
    onModelChanged: PropTypes.func,
    onConfigurationChanged: PropTypes.func,
  };

  onModelChanged = (model) => {
    const { onModelChanged } = this.props;

    return onModelChanged(model);
  };

  onChangeRubricType = (e) => {
    const { model } = this.props;

    const rubricType = e.target.value;
    this.onModelChanged({ ...model, rubricType });
  };

  render() {
    const { model, configuration, canUpdateModel } = this.props;

    const { rubrics = {} } = model || {};
    let { rubricType } = model;
    const { contentDimensions = {}, rubricOptions = [], multiTraitRubric, simpleRubric, rubricless, width } = configuration;
    let rubricTag = '';

    if (!rubricType) {
      rubricType = RUBRIC_TYPES.SIMPLE_RUBRIC;
    }

    const isRubricTypeAvailable =  rubricOptions.indexOf(rubricType) > -1;

    if (canUpdateModel) {
      switch (rubricType) {
        case RUBRIC_TYPES.SIMPLE_RUBRIC:
        default:
          rubricTag = (
            <rubric-configure
              id="simpleRubric"
              key="simple-rubric"
              ref={(ref) => {
                if (ref) {
                  this.simpleRubric = ref;
                  this.simpleRubric.model = {...rubrics.simpleRubric, errors: model.errors || {}};
                  this.simpleRubric.configuration = { ...simpleRubric, width };
                }
              }}
            />
          );
          break;

        case RUBRIC_TYPES.MULTI_TRAIT_RUBRIC:
          rubricTag = (
            <multi-trait-rubric-configure
              id="multiTraitRubric"
              key="multi-trait-rubric"
              ref={(ref) => {
                if (ref) {
                  this.multiTraitRubric = ref;

                  this.multiTraitRubric.model = {...rubrics.multiTraitRubric, errors: model.errors || {} };
                  this.multiTraitRubric.configuration = { ...multiTraitRubric, width: width || multiTraitRubric.width };
                }
              }}
            />
          );
          break;

        case RUBRIC_TYPES.RUBRICLESS:
          rubricTag = (
              <rubric-configure
                  id="rubricless"
                  key="rubricless"
                  ref={(ref) => {
                    if (ref) {
                      this.rubricless = ref;

                      this.rubricless.model = rubrics.rubricless;
                      this.rubricless.configuration = { ...rubricless, width };
                    }
                  }}
              />
          );
          break;
      }
    }

    return (
      <layout.ConfigLayout dimensions={contentDimensions} hideSettings={true} settings={null}>
        <RadioGroup
          aria-label="rubric-type"
          name="rubricType"
          value={model.rubricType}
          onChange={this.onChangeRubricType}
        >
          {
            rubricOptions.length > 1 && rubricOptions.map((availableRubric, i)=>
              <FormControlLabel
                key={i}
                value={availableRubric}
                control={<Radio checked={rubricType === availableRubric} />}
                label={rubricLabels[availableRubric]}
            />
          )
          }
        </RadioGroup>

        {isRubricTypeAvailable && rubricTag}
      </layout.ConfigLayout>
    );
  }
}

export default withStyles(styles)(Main);
