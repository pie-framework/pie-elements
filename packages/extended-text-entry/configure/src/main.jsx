import React from 'react';
import {
  NumberTextField,
  FeedbackSelector,
  InputContainer,
  settings,
  layout
} from '@pie-lib/config-ui';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import { withStyles } from '@material-ui/core/styles';
import EditableHtml from '@pie-lib/editable-html';

const { Panel, toggle } = settings;

const defaultFeedback = {
  type: 'default',
  default: 'Your answer has been submitted'
};

export class Main extends React.Component {
  static propTypes = {
    onModelChanged: PropTypes.func.isRequired,
    onConfigurationChanged: PropTypes.func,
    model: PropTypes.object.isRequired,
    configuration: PropTypes.object.isRequired,
    imageSupport: PropTypes.object.isRequired,
    classes: PropTypes.object.isRequired
  };

  constructor(props) {
    super(props);
    this.state = {
      setDimensions: true
    };
    this.changeWidth = this.change('width');
    this.changeHeight = this.change('height');
  }

  change = key => (event, v) => {
    const { onModelChanged } = this.props;
    const model = this.applyUpdate({ [key]: v });
    onModelChanged(model);
  };

  onPromptChange = markup => {
    const { onModelChanged } = this.props;
    const model = this.applyUpdate({ prompt: markup });
    onModelChanged(model);
  };

  applyUpdate(update) {
    const { model } = this.props;
    const out = Object.assign({}, model, update);
    return out;
  }

  changeFeedback = feedback => {
    const { model, onModelChanged } = this.props;
    const update = { ...model, feedback };
    onModelChanged(update);
  };

  changeTeacherInstructions = teacherInstructions => {
    const { model, onModelChanged } = this.props;
    const update = { ...model, teacherInstructions };
    onModelChanged(update);
  };

  render() {
    const { model, classes, onModelChanged, configuration, onConfigurationChanged, imageSupport } = this.props;
    const {
      equationEditor,
      multiple,
      teacherInstructions = {},
      studentInstructions,
      mathInput,
      width,
      height
    } = configuration;

    return (
      <layout.ConfigLayout
        settings={
          <Panel
            model={model}
            configuration={configuration}
            onChangeModel={model => onModelChanged(model)}
            onChangeConfiguration={config => onConfigurationChanged(config)}
            groups={{
              'Settings': {
                mathInput: mathInput.settings && toggle(mathInput.label),
                'equationEditor.enabled': equationEditor.settings &&
                toggle(equationEditor.label, true),
                'multiple.enabled': multiple.settings &&
                toggle(multiple.label, true),
              },
              'Properties': {
                'teacherInstructions.enabled': teacherInstructions.settings &&
                toggle(teacherInstructions.label, true),
                'configure.studentInstructions.enabled': studentInstructions.settings &&
                toggle(studentInstructions.label),
              },
            }}
          />
        }
      >
        <div>
          <Typography className={classes.header} variant="subheading">
            Display
          </Typography>

          {teacherInstructions.enabled && (
            <InputContainer label={teacherInstructions.label} className={classes.promptContainer}>
              <EditableHtml
                className={classes.prompt}
                markup={model.teacherInstructions || ''}
                onChange={this.changeTeacherInstructions}
                imageSupport={imageSupport}
                nonEmpty={false}
              />
            </InputContainer>
          )}

          {width.settings &&
            <NumberTextField
              label={width.label}
              disabled={!this.state.setDimensions}
              value={parseInt(model.width)}
              min={100}
              max={500}
              onChange={this.changeWidth}
              showErrorWhenOutsideRange
              className={classes.field}
            />
          }
          {
            height.settings &&
              <NumberTextField
                label={height.label}
                disabled={!this.state.setDimensions}
                value={parseInt(model.height)}
                min={100}
                max={500}
                onChange={this.changeHeight}
                showErrorWhenOutsideRange
                className={classes.field}
              />
          }
          <br />
          <InputContainer label="Prompt" className={classes.promptContainer}>
            <EditableHtml markup={model.prompt} onChange={this.onPromptChange} />
          </InputContainer>

          <Typography className={classes.header} variant="subheading">
            Feedback
          </Typography>

          <FeedbackSelector
            label="When submitted, show"
            feedback={model.feedback || defaultFeedback}
            onChange={this.changeFeedback}
          />
        </div>
      </layout.ConfigLayout>
    );
  }
}
export default withStyles(theme => ({
  header: {
    paddingBottom: theme.spacing.unit
  },
  promptContainer: {
    paddingTop: theme.spacing.unit * 2,
    marginBottom: theme.spacing.unit * 2,
    width: '100%'
  },
  promptInput: {
    width: '100%',
    marginBottom: theme.spacing.unit
  },
  field: {
    width: '200px'
  }
}))(Main);
