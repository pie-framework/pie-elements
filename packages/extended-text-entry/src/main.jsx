import React from 'react';
import {EditableHtml} from '@pie-lib/pie-toolbox/editable-html';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import { withStyles } from '@material-ui/core/styles';
import debug from 'debug';
import debounce from 'lodash/debounce';
import { color, Feedback, Collapsible, PreviewPrompt } from '@pie-lib/pie-toolbox/render-ui';
import classnames from 'classnames';

const log = debug('@pie-ui:extended-text-entry');

const style = (theme) => ({
  main: {
    backgroundColor: color.background(),
    color: color.text(),
  },
  prompt: {
    width: '100%',
    color: color.text(),
    marginBottom: theme.spacing.unit * 2,
    fontSize: 'inherit',
  },
  teacherInstructions: {
    marginBottom: theme.spacing.unit * 2,
  },
  editor: {
    marginBottom: theme.spacing.unit * 2,
    borderRadius: '4px',
  },
  srOnly: {
    position: 'absolute',
    left: '-10000px',
    top: 'auto',
    width: '1px',
    height: '1px',
    overflow: 'hidden',
  },
});

export class Main extends React.Component {
  static propTypes = {
    onChange: PropTypes.func.isRequired,
    model: PropTypes.object,
    classes: PropTypes.object.isRequired,
    session: PropTypes.shape({
      value: PropTypes.string,
    }).isRequired,
  };

  changeSession = debounce(this.props.onChange, 1500);

  render() {
    const { model, classes, session } = this.props;

    const {
      dimensions,
      disabled,
      feedback,
      teacherInstructions,
      mathInput,
      spanishInput,
      specialInput,
      animationsDisabled,
      playersToolbarPosition,
      spellCheckEnabled,
    } = model;
    const { value } = session;
    const { width, height } = dimensions || {};
    const maxHeight = '40vh';
    const toolbarOpts = { position: playersToolbarPosition === 'top' ? 'top' : 'bottom' };

    log('[render] disabled? ', disabled);

    const teacherInstructionsDiv = (
      <PreviewPrompt defaultClassName="teacher-instructions" prompt={teacherInstructions} />
    );

    const languageCharactersProps = [];

    if (spanishInput) {
      languageCharactersProps.push({ language: 'spanish' });
    }

    if (specialInput) {
      languageCharactersProps.push({ language: 'special' });
    }

    return (
      <div
        className={classes.main}
        ref={(ref) => {
          this.containerRef = ref;
        }}
      >
        <h2 className={classes.srOnly}>Constructed Response Question</h2>

        {teacherInstructions && (
          <div className={classes.teacherInstructions}>
            {!animationsDisabled ? (
              <Collapsible
                labels={{ hidden: 'Show Teacher Instructions', visible: 'Hide Teacher Instructions' }}
                className={classes.collapsible}
              >
                {teacherInstructionsDiv}
              </Collapsible>
            ) : (
              teacherInstructionsDiv
            )}
          </div>
        )}

        {model.prompt && (
          <Typography component={'span'} className={classes.prompt}>
            <PreviewPrompt defaultClassName="prompt" prompt={model.prompt} />
          </Typography>
        )}

        <EditableHtml
          className={classnames(classes.editor, 'response-area-editor')}
          onChange={this.changeSession}
          markup={value || ''}
          width={width && width.toString()}
          minHeight={height && height.toString()}
          maxHeight={maxHeight}
          disabled={disabled}
          highlightShape={true}
          toolbarOpts={toolbarOpts}
          spellCheck={spellCheckEnabled}
          charactersLimit={50000}
          pluginProps={{
            math: {
              disabled: !mathInput,
              customKeys: this.props.model.customKeys,
              keypadMode: this.props.model.equationEditor,
              controlledKeypadMode: false,
            },
            video: {
              disabled: true,
            },
            audio: {
              disabled: true,
            },
          }}
          languageCharactersProps={languageCharactersProps}
        />

        {feedback && <Feedback correctness="correct" feedback={feedback} />}
      </div>
    );
  }
}

export default withStyles(style)(Main);
