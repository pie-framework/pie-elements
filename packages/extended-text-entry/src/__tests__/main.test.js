import React from 'react';
import { Main } from '../main';
import { shallow } from 'enzyme';

describe('Render Main Component', () => {
  let model = { width: 100, height: 50 };

  it('should match snapshot', () => {
    let wrapper = shallow(
      <Main
        model={model}
        session={{ value: 'hi' }}
        onChange={() => {
        }}
        classes={{}}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot with teacher instructions', () => {
    let wrapper = shallow(
      <Main
        model={{ ...model, teacherInstructions: 'Teacher Instructions' }}
        session={{ value: 'hi' }}
        onChange={() => {
        }}
        classes={{}}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

});
