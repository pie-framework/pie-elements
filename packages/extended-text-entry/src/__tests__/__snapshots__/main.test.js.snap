// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Main Component should match snapshot 1`] = `
<div>
  <EditableHtml
    highlightShape={true}
    markup="hi"
    maxHeight="40vh"
    onChange={[Function]}
    pluginProps={
      Object {
        "audio": Object {
          "disabled": true,
        },
        "math": Object {
          "controlledKeypadMode": false,
          "customKeys": undefined,
          "disabled": true,
          "keypadMode": undefined,
        },
        "video": Object {
          "disabled": true,
        },
      }
    }
  />
</div>
`;

exports[`Render Main Component should match snapshot with teacher instructions 1`] = `
<div>
  <div>
    <WithStyles(Collapsible)
      labels={
        Object {
          "hidden": "Show Teacher Instructions",
          "visible": "Hide Teacher Instructions",
        }
      }
    >
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "Teacher Instructions",
          }
        }
      />
    </WithStyles(Collapsible)>
    <br />
  </div>
  <EditableHtml
    highlightShape={true}
    markup="hi"
    maxHeight="40vh"
    onChange={[Function]}
    pluginProps={
      Object {
        "audio": Object {
          "disabled": true,
        },
        "math": Object {
          "controlledKeypadMode": false,
          "customKeys": undefined,
          "disabled": true,
          "keypadMode": undefined,
        },
        "video": Object {
          "disabled": true,
        },
      }
    }
  />
</div>
`;
